recipes-private/nsls2ptycho/build.sh

```diff
@@ -0,0 +1,19 @@
+# TODO: git clone submodule
+# TODO: detect CUDA, MPI, and FFTW. See below --- 
+# For CUDA, three issues: 1. If users want to run the GPU version, CUDA
+# driver and toolkit must exist. The latter might be taken care of by conda, 
+# but users must handle the former. 2. Conda has no up-to-date CuPy, and 
+# users are encouraged to pip install a pre-built one based on the toolkit
+# version, which makes the automation a bit more difficult. 3. User may only 
+# need the CPU version, then we don't need to touch anything GPU-related!
+#
+# For MPI (and mpi4py), the problem is to build mpi4py on top of whatever
+# MPI in the system. Conda is unable to do a customized build, and it's 
+# better to install a specified MPI first (either from Conda or by oneself,
+# if, say, cuda-aware MPI is needed), and then pip install mpi4py.
+#
+# Currently this script assumes 1. CuPy is either not needed or installed 
+# separately, and 2. "conda install mpi4py" works. Obviously we need to 
+# polish this when more beamlines need this pacakge...
+
+$PYTHON -m pip install . --no-deps --ignore-installed -vv
```

Comment:
If it's the only command needed, it can be handled via `meta.yml`. The comments are valuable, but can be in a `README.txt` or something similar...

Reply:
This belongs to a discussion with a bigger scope, maybe worthy of a separate issue. I tend to leave the script here until I (or we) find a satisfying solution for the dependency issue. 

First, non-Python libraries (ex: HDF5, FFTW) can be built against MPI to provide asynchronous processing. Then, the corresponding Python wrappers (ex: h5py, PyFFTW) need to be built against those specialized versions. It means at the Conda level the complexity inflates quickly, and I am not sure what is the best strategy (use build string as unique id? use different output names?) to handle this.

Second, when it comes to supporting (NVIDIA) GPUs, even MPI itself can have different flavors, so called CUDA-awareness. Then, in addition to building mpi4py against a CUDA-aware version, for example, do we build HDF5 against a CUDA-aware MPI or not? Too many degrees of freedom come into play, and so far we only keep the minimum requirement. 

I feel that eventually a Bash or Python script will be needed to help Conda resolve this issue, especially in the coming Jupyter-SDCC era. The `meta.yaml` recipe alone might not be enough.

Reply:
OK, sounds good. Let's keep it as a placeholder for the future complications then.